src/navigation/_AppSettingsProvider.jsx

```diff
@@ -0,0 +1,76 @@
+import React, {
+  useState,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import AppSettingsContext from './_AppSettingsContext';
+import { settingsConfigPropType } from '../site/siteConfigPropTypes';
+
+const propTypes = {
+  children: PropTypes.element.isRequired,
+  settingsConfig: settingsConfigPropType.isRequired,
+};
+
+const AppSettingsProvider = ({ settingsConfig, children }) => {
+  const {
+    defaultLocale = 'en',
+    defaultTheme,
+    defaultDirection = 'ltr',
+    themes,
+  } = settingsConfig;
+  const [state, setState] = useState({
```

Comment:
Could we flatten these out to their own state const?
```
 const [locale, setLocale] = useState(defaultLocale)
```

Reply:
yes